activerecord/test/cases/strict_loading_test.rb

```diff
@@ -298,6 +298,26 @@ def test_strict_loading_with_has_many_singular_association_and_reload
     end
   end
 
+  def test_strict_loading_with_nested_association_reload
+    with_strict_loading_by_default(Project) do
+      with_strict_loading_by_default(Developer) do
+        with_strict_loading_by_default(AuditLog) do
```

Comment:
It is worth extending the method for multiple args:

```suggestion
    with_strict_loading_by_default(Project, Developer, AuditLog) do
```